app/lib/actions/katello/product/destroy.rb

```diff
@@ -55,6 +55,7 @@ def plan(product, options = {})
 
         def finalize
           product = ::Katello::Product.find(input[:product_id])
+          ::Katello::Subscription.where(:cp_id => product.cp_id, :organization_id => product.organization_id).destroy_all
```

Comment:
why was this needed justin?

Reply:
Did you intend to put a `dependent: :destroy` on K::Pool subscriptions?